### ISSUE TITLE
dbnet++  Poor detection of long text

### ISSUE THREAD

**Issue by qiuzhixin9527** [open-mmlab/mmocr#1869]
Bad performance on long text detection, 
![image](https://user-images.githubusercontent.com/37932939/233298999-9a36a8be-9dc6-45de-8a7b-9ba3e056c925.png)
![image](https://user-images.githubusercontent.com/37932939/233299606-0f810a76-43d9-4850-9661-544b8d86e771.png)
![image](https://user-images.githubusercontent.com/37932939/233300002-f92ce264-864c-49b5-841f-01977a0c97c7.png)

it is my config
model = dict(
    type='DBNet',
    backbone=dict(
        type='mmdet.ResNet',
        depth=50,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=-1,
        norm_cfg=dict(type='BN', requires_grad=True),
        norm_eval=False,
        style='pytorch',
        dcn=dict(type='DCNv2', deform_groups=1, fallback_on_stride=False),
        init_cfg=dict(type='Pretrained', checkpoint='torchvision://resnet50'),
        stage_with_dcn=(False, True, True, True)),
    neck=dict(
        type='FPNC',
        in_channels=[256, 512, 1024, 2048],
        lateral_channels=256,
        asf_cfg=dict(attention_type='ScaleChannelSpatial')),
    det_head=dict(
        type='DBHead',
        in_channels=256,
        module_loss=dict(type='DBModuleLoss'),
        postprocessor=dict(
            type='DBPostprocessor', text_repr_type='quad',
            epsilon_ratio=0.002)),
    data_preprocessor=dict(
        type='TextDetDataPreprocessor',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        bgr_to_rgb=True,
        pad_size_divisor=32))
train_pipeline = [
    dict(type='LoadImageFromFile', color_type='color_ignore_orientation'),
    dict(
        type='LoadOCRAnnotations',
        with_bbox=True,
        with_polygon=True,
        with_label=True),
    dict(
        type='TorchVisionWrapper',
        op='ColorJitter',
        brightness=0.12549019607843137,
        saturation=0.5),
    dict(
        type='ImgAugWrapper',
        args=[['Fliplr', 0.5], {
            'cls': 'Affine',
            'rotate': [-10, 10]
        }, ['Resize', [0.5, 3.0]]]),
    dict(type='RandomCrop', min_side_ratio=0.1),
    dict(type='Resize', scale=(1024, 1024), keep_ratio=True),
    dict(type='Pad', size=(1024, 1024)),
    dict(
        type='PackTextDetInputs',
        meta_keys=('img_path', 'ori_shape', 'img_shape'))
]
test_pipeline = [
    dict(type='LoadImageFromFile', color_type='color_ignore_orientation'),
    dict(type='Resize', scale=(1024, 1024), keep_ratio=True),
    dict(
        type='LoadOCRAnnotations',
        with_polygon=True,
        with_bbox=True,
        with_label=True),
    dict(
        type='PackTextDetInputs',
        meta_keys=('img_path', 'ori_shape', 'img_shape', 'scale_factor',
                   'instances'))
]
default_scope = 'mmocr'
env_cfg = dict(
    cudnn_benchmark=False,
    mp_cfg=dict(mp_start_method='fork', opencv_num_threads=0),
    dist_cfg=dict(backend='nccl'))
randomness = dict(seed=None)
default_hooks = dict(
    timer=dict(type='IterTimerHook'),
    logger=dict(type='LoggerHook', interval=5),
    param_scheduler=dict(type='ParamSchedulerHook'),
    checkpoint=dict(type='CheckpointHook', interval=20),
    sampler_seed=dict(type='DistSamplerSeedHook'),
    sync_buffer=dict(type='SyncBuffersHook'),
    visualization=dict(
        type='VisualizationHook',
        interval=1,
        enable=False,
        show=False,
        draw_gt=False,
        draw_pred=False))
log_level = 'INFO'
log_processor = dict(type='LogProcessor', window_size=10, by_epoch=True)
load_from = 'https://download.openmmlab.com/mmocr/textdet/dbnetpp/tmp_1.0_pretrain/dbnetpp_r50dcnv2_fpnc_100k_iter_synthtext-20220502-352fec8a.pth'
resume = False
val_evaluator = dict(type='HmeanIOUMetric')
test_evaluator = dict(type='HmeanIOUMetric')
vis_backends = [dict(type='LocalVisBackend')]
visualizer = dict(
    type='TextDetLocalVisualizer',
    name='visualizer',
    vis_backends=[dict(type='LocalVisBackend')])
icdar2015_textdet_data_root = '/home/aipf/work/团队共享目录/zhixin/datasets/icdar2015'
icdar2015_textdet_train = dict(
    type='OCRDataset',
    data_root='/home/aipf/work/团队共享目录/zhixin/datasets/icdar2015',
    ann_file='textdet_train.json',
    filter_cfg=dict(filter_empty_gt=True, min_size=32),
    pipeline=None)
icdar2015_textdet_test = dict(
    type='OCRDataset',
    data_root='/home/aipf/work/团队共享目录/zhixin/datasets/icdar2015',
    ann_file='textdet_test.json',
    test_mode=True,
    pipeline=None)
optim_wrapper = dict(
    type='OptimWrapper',
    optimizer=dict(type='SGD', lr=0.0035, momentum=0.9, weight_decay=0.0001))
train_cfg = dict(type='EpochBasedTrainLoop', max_epochs=1200, val_interval=20)
val_cfg = dict(type='ValLoop')
test_cfg = dict(type='TestLoop')
param_scheduler = [dict(type='PolyLR', power=0.9, eta_min=1e-07, end=200)]
train_list = [
    dict(
        type='OCRDataset',
        data_root='/home/aipf/work/团队共享目录/zhixin/datasets/icdar2015',
        ann_file='textdet_train.json',
        filter_cfg=dict(filter_empty_gt=True, min_size=32),
        pipeline=None)
]
test_list = [
    dict(
        type='OCRDataset',
        data_root='/home/aipf/work/团队共享目录/zhixin/datasets/icdar2015',
        ann_file='textdet_test.json',
        test_mode=True,
        pipeline=None)
]
train_dataloader = dict(
    batch_size=8,
    num_workers=8,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type='ConcatDataset',
        datasets=[
            dict(
                type='OCRDataset',
                data_root='/home/aipf/work/团队共享目录/zhixin/datasets/icdar2015',
                ann_file='textdet_train.json',
                filter_cfg=dict(filter_empty_gt=True, min_size=32),
                pipeline=None)
        ],
        pipeline=[
            dict(
                type='LoadImageFromFile',
                color_type='color_ignore_orientation'),
            dict(
                type='LoadOCRAnnotations',
                with_bbox=True,
                with_polygon=True,
                with_label=True),
            dict(
                type='TorchVisionWrapper',
                op='ColorJitter',
                brightness=0.12549019607843137,
                saturation=0.5),
            dict(
                type='ImgAugWrapper',
                args=[['Fliplr', 0.5], {
                    'cls': 'Affine',
                    'rotate': [-10, 10]
                }, ['Resize', [0.5, 3.0]]]),
            dict(type='RandomCrop', min_side_ratio=0.1),
            dict(type='Resize', scale=(1024, 1024), keep_ratio=True),
            dict(type='Pad', size=(1024, 1024)),
            dict(
                type='PackTextDetInputs',
                meta_keys=('img_path', 'ori_shape', 'img_shape'))
        ]))
val_dataloader = dict(
    batch_size=8,
    num_workers=8,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type='ConcatDataset',
        datasets=[
            dict(
                type='OCRDataset',
                data_root='/home/aipf/work/团队共享目录/zhixin/datasets/icdar2015',
                ann_file='textdet_test.json',
                test_mode=True,
                pipeline=None)
        ],
        pipeline=[
            dict(
                type='LoadImageFromFile',
                color_type='color_ignore_orientation'),
            dict(type='Resize', scale=(1024, 1024), keep_ratio=True),
            dict(
                type='LoadOCRAnnotations',
                with_polygon=True,
                with_bbox=True,
                with_label=True),
            dict(
                type='PackTextDetInputs',
                meta_keys=('img_path', 'ori_shape', 'img_shape',
                           'scale_factor', 'instances'))
        ]))
test_dataloader = dict(
    batch_size=8,
    num_workers=8,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type='ConcatDataset',
        datasets=[
            dict(
                type='OCRDataset',
                data_root='/home/aipf/work/团队共享目录/zhixin/datasets/icdar2015',
                ann_file='textdet_test.json',
                test_mode=True,
                pipeline=None)
        ],
        pipeline=[
            dict(
                type='LoadImageFromFile',
                color_type='color_ignore_orientation'),
            dict(type='Resize', scale=(1024, 1024), keep_ratio=True),
            dict(
                type='LoadOCRAnnotations',
                with_polygon=True,
                with_bbox=True,
                with_label=True),
            dict(
                type='PackTextDetInputs',
                meta_keys=('img_path', 'ori_shape', 'img_shape',
                           'scale_factor', 'instances'))
        ]))
launcher = 'none'
work_dir = 'output/dbpp0417_dcnv2'

_Originally posted by @qiuzhixin9527 in https://github.com/open-mmlab/mmocr/issues/376#issuecomment-1515893519_

**Comment by YuboLong** [comment]
I'm using dbnet . I have the same problem, especially when the a text line is almost the same compare to the image width

**Comment by Mountchicken** [comment]
Hi @qiuzhixin9527 @YuboLong 

![45351679471024_ pic](https://user-images.githubusercontent.com/65173622/236107882-4d7679ae-1cac-4575-adcb-5eb924376116.jpg)

We trained DBNet++ on [HierText](https://github.com/google-research-datasets/hiertext) to conduct text line detection, and it somehow performs well. DBNet indeed performs less satisfying on long text, due to its text shrinking strategy as discussed in [https://github.com/open-mmlab/mmocr/issues/376](https://github.com/open-mmlab/mmocr/issues/376). However, you can try to use more training data and see if that helps (Hiertext provides around 8000 images for text line detection)

**Comment by alexwang42** [comment]
Hi! @Mountchicken, ths for you reply. I also trained DBNet++ on my private chinese  dataset. It performs well on long text line. But I notice the textbox is somehow narrow than my annotation box like your result:
![image](https://github.com/open-mmlab/mmocr/assets/96226065/6d93f372-878f-4275-9da9-33ebecbd4e91). 
How can I get the textbox as wide as my annotation box? Keep traning? Or just I use some postprocessing on my detection result? Sorry for my poor english.

**Comment by YuboLong** [comment]
> Hi! @Mountchicken, ths for you reply. I also trained DBNet++ on my private chinese dataset. It performs well on long text line. But I notice the textbox is somehow narrow than my annotation box like your result: ![image](https://user-images.githubusercontent.com/96226065/250432378-6d93f372-878f-4275-9da9-33ebecbd4e91.png). How can I get the textbox as wide as my annotation box? Keep traning? Or just I use some postprocessing on my detection result? Sorry for my poor English.

I personally scaled the predict boxes on my own postprocess code , different width/height ratio boxes have different scale rate.
There is a ZhiHu post talk about it , you can refer it here [DBNet GT 构造过程的问题和对应解决方法](https://zhuanlan.zhihu.com/p/579956622)

**Comment by alexwang42** [comment]
> > Hi! @Mountchicken, ths for you reply. I also trained DBNet++ on my private chinese dataset. It performs well on long text line. But I notice the textbox is somehow narrow than my annotation box like your result: ![image](https://user-images.githubusercontent.com/96226065/250432378-6d93f372-878f-4275-9da9-33ebecbd4e91.png). How can I get the textbox as wide as my annotation box? Keep traning? Or just I use some postprocessing on my detection result? Sorry for my poor English.
> 
> I personally scaled the predict boxes on my own postprocess code , different width/height ratio boxes have different scale rate. There is a ZhiHu post talk about it , you can refer it here [DBNet GT 构造过程的问题和对应解决方法](https://zhuanlan.zhihu.com/p/579956622)

Appreciate. I'll check the post carefully.
